### PR TITLE
Add instrument search service - popular and keyword search [stage-8]

### PR DESCRIPTION
**Pull request by stulees**
- Adding `instrumentSearchService` to retrieve instruments
  -  _popularSearch_  returns most common instruments based on _category_ 
  - _keywordSearch_ returns any instruments that match _keyword_ and _category_

### CODE REVIEW

File: web/scripts/template-editor/services/svc-instrument-search.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        results = {
+          keyword: {},
+          popular: {}
+        };
+
+      factory.keywordSearch = function( category, keyword ) {
+        var keywordProp = category + "|" + keyword;
+
+        if ( results.keyword[ keywordProp ] ) {
+          return $q.when( results.keyword[ keywordProp ] );
```
Comment (santiagonoguez) [inline web/scripts/template-editor/services/svc-instrument-search.js]: @stulees this is a cache, right ?

Comment (stulees) [reply web/scripts/template-editor/services/svc-instrument-search.js]: @santiagonoguez That's correct

**Comment by fjvallarino** [comment]
@stulees looks good 👍

File: web/scripts/template-editor/services/svc-instrument-search.js (hunk 1 above)
Comment (fjvallarino) [reply web/scripts/template-editor/services/svc-instrument-search.js]: @stulees should we consider expiring this cache at some point, or do the instrument prices last long enough for this to be not needed?

Comment (stulees) [reply web/scripts/template-editor/services/svc-instrument-search.js]: @fjvallarino The instruments returned in these endpoint calls do not return actual financial data for the instruments, that is a separate (license based) financial request that is facilitated by our `rise-data-financial` web component in the HTML template page. The instruments returned in these endpoint calls here will not change frequently and therefore we can just rely on caching in memory to reduce the amount of endpoint calls to make.

**Pull request merged**